### PR TITLE
John conroy/ws access CAT-257

### PR DESCRIPTION
**Pull request by john-conroy**

### CODE REVIEW

File: context/app/static/js/components/Header/UserLinks/UserLinks.jsx [hunk 1]
```diff
@@ -10,13 +11,14 @@ import { TruncatedSpan, WarningDropdownLink } from './style';
...
 
+  const userType = getUserType();
   return (
     <Dropdown
       title={<TruncatedSpan>{isAuthenticated ? userEmail || 'User' : 'User Profile'}</TruncatedSpan>}
       data-testid="user-profile-dropdown"
     >
       <DropdownLink href="/my-lists">My Lists</DropdownLink>
-      {isWorkspacesUser && <DropdownLink href="/workspaces">My Workspaces</DropdownLink>}
+      {(isWorkspacesUser || userType === 'internal') && <DropdownLink href="/workspaces">My Workspaces</DropdownLink>}
```
Comment (NickAkhmetov) [inline context/app/static/js/components/Header/UserLinks/UserLinks.jsx:21]: Perhaps adjusting the "isWorkspacesUser" logic in the app context instead could help avoid unexpected regressions in other places?

Comment (john-conroy) [reply context/app/static/js/components/Header/UserLinks/UserLinks.jsx:21]: We are only changing the behavior for the dropdown to be available to workspace users or users who have authenticated in the past (also covers currently authenticated, non-workspace users). The auth guard will behave the same as it has. Maybe I'm misinterpreting your comment.

Comment (john-conroy) [reply context/app/static/js/components/Header/UserLinks/UserLinks.jsx]: Added a var to make this more clear per your suggestion.

**Review by NickAkhmetov — DISMISSED** [review]
Thanks for clearing up the confusion 🙏

**Review by NickAkhmetov — APPROVED** [review]

**Pull request merged**